packages/rum-core/src/performance-monitoring/perf-entry-recorder.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { LONG_TASK, LARGEST_CONTENTFUL_PAINT } from '../common/constants'
+import { noop } from '../common/utils'
+import Span from './span'
+
+/**
+ * Create Spans for the long task entries
+ * Spec - https://w3c.github.io/longtasks/
+ */
+function createLongTaskSpans(longtasks) {
+  const spans = []
+
+  for (let i = 0; i < longtasks.length; i++) {
+    /**
+     * name of the long task can be any one of the type listed here
+     * depeding on the origin of the longtask
+     * https://w3c.github.io/longtasks/#sec-PerformanceLongTaskTiming
+     */
+    const { name, startTime, duration, attribution } = longtasks[i]
+    const end = startTime + duration
+    const kind = LONG_TASK
```

Comment:
I am also not sure though, We would need like a generic one something like `browser-events`. then we can simply group them all in to this category instead of expanded ones currently like `hard-navigation`, `resource`, `measure` and so on.

Reply:
The reason why I did not put it as `app` is also because these can come from third party scripts and also ad's. Just FYI

Reply:
I think `longtask` is too specific, we probably should group these kind of spans under something like `browser-metric` or `perf-metric`. These are generic enough that we can use them for spans that don't fit into other kinds.

Reply:
i remember it now, the reason why it is `longtask` was because we calculate breakdowns based on span type and having them as longtasks would give an advantage to user that % of transaction is spent on long tasks vs if we use `browser-metric or events` its not super helpful. WDYT?